### PR TITLE
[BIOMAGE-1609] Test mito and doublet score

### PR DESCRIPTION
**Pull request by alexvpickering**
# Background
#### Link to issue 

https://biomage.atlassian.net/jira/software/c/projects/BIOMAGE/boards/1?modal=detail&selectedIssue=BIOMAGE-1606

#### Link to staging deployment URL 

https://ui-alex-worker226.scp-staging.biomage.net/

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [x] Unit tests written
- [x] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [x] Deployed to staging
- [ ] Passed integration tests 

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: r/R/get_metadata_information.R [hunk 1]
```diff
@@ -30,26 +14,33 @@ getDoubletScore <- function(req, data) {
 # the MT-genes only in MMusculus and Homo Sapiens by grepping "MT"
```
Comment (gerbeldo) [inline r/R/get_metadata_information.R]:
```suggestion
# the MT-genes only in Mus musculus, Homo sapiens  or drosophila by grepping "MT[-:]"
```

File: r/R/get_metadata_information.R [hunk 2]
```diff
@@ -30,26 +14,33 @@ getDoubletScore <- function(req, data) {
...
-    if(any(is.na(result)))
-        warning("There are missing values in the doublet_scores results")
+  # get the specified column, ordering by cells_id
+  cells_id_order <- order(data$cells_id, decreasing = FALSE)
+  result <- data@meta.data[cells_id_order, column]
+  result<- as.data.frame(result)
+  result$cells_id <- data@meta.data$cells_id[cells_id_order]
+  result <- result %>%
+    tidyr::complete(cells_id = seq(0, max(data@meta.data$cells_id))) %>%
+    select(-cells_id)
```
Comment (gerbeldo) [inline r/R/get_metadata_information.R]:
```suggestion
    dplyr::select(-cells_id)
```

File: r/tests/testthat/test-basic.R [hunk 3]
```diff
@@ -1,3 +0,0 @@
-test_that("multiplication works", {
-  expect_equal(2 * 2, 4)
-})
```
Comment (gerbeldo) [inline r/tests/testthat/test-basic.R:3]: what if multiplication stops working? 🤔

File: r/R/get_metadata_information.R [hunk 4]
```diff
@@ -30,26 +14,33 @@ getDoubletScore <- function(req, data) {
...
+  result <- t(unname(result))
```
Comment (gerbeldo) [inline r/R/get_metadata_information.R]: this throws warnings (18 of them) when running tests. `The value argument of names<- can't be empty as of tibble 3.0.0. Column 1 must be named.`

If we care about this, we could "declass" to `"data.frame"` using `as.data.frame` or `class(result) <- "data.frame"`

```suggestion
  result <- t(unname(as.data.frame(result)))
```
If we don't, just leave as is.

File: r/R/get_metadata_information.R [hunk 5]
```diff
@@ -4,52 +4,43 @@
...
+    stop(column, " is not computed for this experiment.")
+  }
 
-    # Be aware of possible na values
-    if(any(is.na(result)))
-        warning("There are missing values in the doublet_scores results")
+  # get the specified column, ordering by cells_id
+  cells_id_order <- order(data$cells_id, decreasing = FALSE)
+  result <- data@meta.data[cells_id_order, column]
+  result<- as.data.frame(result)
```
Comment (alexvpickering) [inline r/R/get_metadata_information.R]:
```suggestion
  result <- as.data.frame(result)
```

**Review by gerbeldo — APPROVED** [review]

**Pull request merged**